src/js/views/maps/LayerDetailsView.js

```diff
@@ -143,197 +134,153 @@ define([
         },
       ],
 
-      /**
-       * Creates an object that gives the events this view will listen to and the
-       * associated function to call. Each entry in the object has the format 'event
-       * selector': 'function'.
-       * @returns {Object}
-       */
-      events: function () {
-        var events = {};
+      /** @inheritdoc */
+      events() {
+        const events = {};
         // Close the layer details panel when the toggle button is clicked. Get the
         // class of the toggle button from the classes property set in this view.
-        events["click ." + this.classes.toggle] = "close";
+        events[`click .${this.classes.toggle}`] = "close";
         return events;
       },
 
       /**
        * Whether or not the layer details view is open
-       * @type {Boolean}
+       * @type {boolean}
        */
       isOpen: false,
 
       /**
        * Executed when a new LayerDetailsView is created
-       * @param {Object} [options] - A literal object with options to pass to the view
+       * @param {object} [options] - A literal object with options to pass to the view
        */
-      initialize: function (options) {
-        try {
-          // Get all the options and apply them to this view
-          if (typeof options == "object") {
-            for (const [key, value] of Object.entries(options)) {
-              this[key] = value;
-            }
-          }
-        } catch (e) {
-          console.log(
-            "A LayerDetailsView failed to initialize. Error message: " + e,
-          );
+      initialize(options) {
+        // Get all the options and apply them to this view
+        if (typeof options === "object") {
+          Object.entries(options).forEach(([key, value]) => {
```

Comment:
I think `Object.assign(this, options)` works in the same way; shorter, just as readable?

Reply:
like! done